src/snapshot-editor/src/edit_vmstate.rs

```diff
@@ -0,0 +1,156 @@
+// Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::path::PathBuf;
+
+use clap::Subcommand;
+use clap_num::maybe_hex;
+use vmm::arch::aarch64::regs::Aarch64RegisterVec;
+use vmm::persist::MicrovmState;
+
+use crate::utils::{open_vmstate, save_vmstate, UtilsError};
+
+#[derive(Debug, thiserror::Error)]
+pub enum EditVmStateError {
+    #[error("{0}")]
+    Utils(#[from] UtilsError),
+}
+
+#[derive(Debug, Subcommand)]
+pub enum EditVmStateSubCommand {
+    /// Remove registers from vcpu states.
+    #[cfg(target_arch = "aarch64")]
+    RemoveRegs {
+        /// Set of regs to remove.
+        #[arg(value_parser=maybe_hex::<u64>, num_args = 1.., value_delimiter = ' ')]
+        regs: Vec<u64>,
+        /// Path to the vmstate file.
+        #[arg(short, long)]
+        vmstate_path: PathBuf,
+        /// Path of output file.
+        #[arg(short, long)]
+        output_path: PathBuf,
+    },
+}
+
+pub fn edit_vmstate_command(command: EditVmStateSubCommand) -> Result<(), EditVmStateError> {
+    match command {
+        #[cfg(target_arch = "aarch64")]
+        EditVmStateSubCommand::RemoveRegs {
+            regs,
+            vmstate_path,
+            output_path,
+        } => edit(&vmstate_path, &output_path, |state| {
+            remove_regs(state, &regs)
+        })?,
+    }
+    Ok(())
+}
+
+fn edit(
+    vmstate_path: &PathBuf,
+    output_path: &PathBuf,
+    f: impl Fn(MicrovmState) -> Result<MicrovmState, EditVmStateError>,
+) -> Result<(), EditVmStateError> {
+    let (microvm_state, version) = open_vmstate(vmstate_path)?;
+    let microvm_state = f(microvm_state)?;
+    save_vmstate(microvm_state, output_path, version)?;
+    Ok(())
+}
+
+#[cfg(target_arch = "aarch64")]
+fn remove_regs(
```

Comment:
what's the point of passing this function pointer in `edit`? It looks to me it's a needless indirection.
Why not simply call `remove_regs` when you get the `EditVmStateSubCommand::RemoveRegs` subcommand?

Reply:
The idea here is to have a common code for all `edit` methods in one method. So if we add new `edit` methods it can be used as such:
```rust
 match command {
        SubCommand::RemoveRegs => edit(&vmstate_path, &output_path, |state| {
            remove_regs(state, &regs)
        })?,
        StateSubCommand::SomeEditMethod2 => edit(&vmstate_path, &output_path, some_edit_method_2)?,
        StateSubCommand::SomeEditMethod3 => edit(&vmstate_path, &output_path, some_edit_method_3)?,
}
```

Reply:
I got it, I just think it's unnecessary indirection. What's the benefit of this over:

```rust
 match command {
        SubCommand::RemoveRegs => remove_regs(&vmstate_path, &output_path)?,
        StateSubCommand::SomeEditMethod2 => some_edit_method_2(&vmstate_path, &output_path)?,
        StateSubCommand::SomeEditMethod3 => some_edit_method_3(&vmstate_path, &output_path)?,
}
```

Reply:
I guess it is just personal preference. Otherwise each edit method will start with ` let (microvm_state, version) = open_vmstate(vmstate_path)?;` and end with `save_vmstate(microvm_state, output_path, version)?;`. Considering how small the amount of code is, both versions can be correct.